File: apps/agent/internal/heartbeat/messaging.go
// Copyright © Advanced Micro Devices, Inc., or its affiliates.
//
// SPDX-License-Identifier: MIT

package heartbeat

import (
	"time"

	"github.com/silogen/agent/internal/messaging"
)

type HeartbeatMessage struct {
	MessageType     messaging.MessageType `json:"message_type"`
	LastHeartbeatAt time.Time             `json:"last_heartbeat_at"`
	ClusterName     string                `json:"cluster_name"`
}

File: apps/api/airm/app/utilities/messaging.py
# Copyright © Advanced Micro Devices, Inc., or its affiliates.
#
# SPDX-License-Identifier: MIT

"""Shared Pydantic pieces for RabbitMQ payloads that embed Kubernetes manifests."""

from pydantic import BaseModel, ConfigDict


class KubernetesMetadata(BaseModel):
    """Kubernetes metadata section."""

    model_config = ConfigDict(extra="allow")  # Allow additional fields like resourceVersion, uid, etc.

    name: str | None = None
    namespace: str | None = None
    labels: dict[str, str] | None = None
    annotations: dict[str, str] | None = None
